### DISCUSSION TITLE
Coole Sache!

### DISCUSSION THREAD

**Discussion by ChriD** [cspiel1/fspctl#6] [Q&A]
Coole Sache!
Sag mal der deamon pusht aber nur die ganzen registerwerte auf mqtt
Hast du da auch irgendwas gemacht bzg. historische daten? Bzw. Wie rechnest du dir dann später die Verbräche / Generierung pro Tag aus? Bin auch gerade dabei das ich für einen bekannten eine Auswertung von seinem FSP machen sollte.

**Comment by cspiel1** [comment]
The most reliable history data can be queried from the converter directly. It is done in `src/fenergy.c` and is published via mqtt. I made  a home assistant page with a `type: custom:apexcharts-card` that visualizes the energy output of the last 7 days, 12 month and for each year. E.g.
![month](https://github.com/cspiel1/fspctl/assets/14180877/1e08298a-93fb-4426-a00e-44674c8a8051)

**Comment by ChriD** [comment]
Thank you for the information!
